### PR TITLE
add the one-line script to install AM collector

### PR DESCRIPTION
**Pull request by jizh6311**
https://aspenmesh.atlassian.net/browse/AM-4664

### CODE REVIEW

File: docs/aspenmesh-collector.MD [hunk 1]
```diff
@@ -4,6 +4,14 @@ The Aspen OpenTelemetry Collector collects telemetry data from Kubernetes and yo
 
 ## Installation
 
+### One-line script installation
+
+The Aspen OpenTelemetry Collector can be installed by a one-line script.
+```bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/aspenmesh/aspenmesh-charts/main/scripts/aspenmesh-collector-setup.sh)" -- --endpoint <remote-write-endpoint> --apiKey <user-group-api-key>
```
Comment (jizh6311) [inline docs/aspenmesh-collector.MD:11]: Unfortunately, `https://git.io/` is no longer accepting new links for shortening URL. Are we OK with the github raw content URL? Is it worth to search another free and legit third-party website?

Comment (jizh6311) [reply docs/aspenmesh-collector.MD:11]: I think `/bin/bash` works for most users? Windows users need to either install `sh` or `bash shell` anyway

File: scripts/aspenmesh-collector-setup.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  echo "helm is not installed. Please visit https://helm.sh/docs/intro/install/" >&2
+  exit 1
+fi
+
+usage() {
+  echo "
+usage: ${0} [flags] [-h|--help]
+Flags               Purpose
+--apiKey            Required; The FusionAuth API key for the user's group
+--endpoint          Optional; The Promscale remote write endpoint. Default value: https://metrics.cloud.aspenmesh.io/write
```
Comment (ricksalsa) [inline scripts/aspenmesh-collector-setup.sh:15]: Let's remove this, as it's not necessary for end users. We only have one endpoint today, which the chart defaults to.

File: scripts/aspenmesh-collector-setup.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+" >&2
+
+    exit 1
+}
+
+apiKey=""
+endpoint="https://metrics.cloud.aspenmesh.io/write"
```
Comment (ricksalsa) [inline scripts/aspenmesh-collector-setup.sh:22]: same here - remove

File: scripts/aspenmesh-collector-setup.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+if ! [ -x "$(command -v helm)" ]; then
+  echo "helm is not installed. Please visit https://helm.sh/docs/intro/install/" >&2
```
Comment (ricksalsa) [inline scripts/aspenmesh-collector-setup.sh:6]: Nit: Helm should be capitalized. Update the message to be:

> Missing dependency: Helm \n This script requires Helm to install the Aspen Mesh distribution for OpenTelemetry. Please visit https://helm.sh/docs/intro/install/ and rerun this script once Helm is installed.

File: scripts/aspenmesh-collector-setup.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+while [[ $# -gt 0 ]]
+do
+  key="$1"
+  case $key in
+    --apiKey)
+    apiKey="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    --endpoint)
```
Comment (ricksalsa) [inline scripts/aspenmesh-collector-setup.sh:32]: Remove `--endpoint`

**Review by ricksalsa — CHANGES_REQUESTED** [review]

File: docs/aspenmesh-collector.MD [hunk 6]
```diff
@@ -4,6 +4,14 @@ The Aspen OpenTelemetry Collector collects telemetry data from Kubernetes and yo
 
 ## Installation
 
+### One-line script installation
+
+The Aspen OpenTelemetry Collector can be installed by a one-line script.
```
Comment (tartale) [inline docs/aspenmesh-collector.MD:9]: I vaguely remember Joanna saying something about us removing "OpenTelemetry" from our collector name? let's check, and do that here if needed

File: docs/aspenmesh-collector.MD [hunk 7]
```diff
@@ -4,6 +4,14 @@ The Aspen OpenTelemetry Collector collects telemetry data from Kubernetes and yo
...
+```bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/aspenmesh/aspenmesh-charts/main/scripts/aspenmesh-collector-setup.sh)" -- --endpoint <remote-write-endpoint> --apiKey <user-group-api-key>
+```
+You can also follow the details for each installation step.
```
Comment (tartale) [inline docs/aspenmesh-collector.MD:13]: suggested wording:

```suggestion
Alternatively, you can install using the following steps
```

File: scripts/aspenmesh-collector-setup.sh [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+if ! [ -x "$(command -v helm)" ]; then
+  echo "helm is not installed. Please visit https://helm.sh/docs/intro/install/" >&2
+  exit 1
+fi
+
+usage() {
+  echo "
+usage: ${0} [flags] [-h|--help]
+Flags               Purpose
+--apiKey            Required; The FusionAuth API key for the user's group
```
Comment (tartale) [inline scripts/aspenmesh-collector-setup.sh:14]: The user wouldn't know what FusionAuth is, so suggest re-phrasing as:
```suggestion
--apiKey            Required; The API key provided upon initial signup to Aspen Mesh
```

File: scripts/aspenmesh-collector-setup.sh (hunk 2 above)
Comment (tartale) [reply scripts/aspenmesh-collector-setup.sh:15]: Let's definitely remove it from the help, but keep the functionality it for internal use

File: scripts/aspenmesh-collector-setup.sh (hunk 3 above)
Comment (tartale) [reply scripts/aspenmesh-collector-setup.sh:22]: We need this for testing

File: scripts/aspenmesh-collector-setup.sh [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    shift # past value
+    ;;
+    *)    # unknown option
+    usage
+    ;;
+  esac
+done
+
+if [[ -z "${apiKey}" ]]; then
+  cho "missing <apiKey> argument" >&2
```
Comment (tartale) [inline scripts/aspenmesh-collector-setup.sh:44]: type-o:
```suggestion
  echo "missing <apiKey> argument" >&2
```

**Review by tartale — APPROVED** [review]
looks good; I'd handle @ricksalsa 's suggestion for endpoint differently though.

File: scripts/aspenmesh-collector-setup.sh (hunk 2 above)
Comment (jizh6311) [reply scripts/aspenmesh-collector-setup.sh:15]: @tartale Rick also left a comment in the story and explained why one AC was to create a new `cli` repo: https://aspenmesh.atlassian.net/browse/AM-4659?focusedCommentId=13091. Thoughts?

Comment (tartale) [reply scripts/aspenmesh-collector-setup.sh:15]: That seems fine to me. The ability to override `--endpoint` is important to me though.

I have a suggestion for that below that makes it less obvious that you can override; see below

File: scripts/aspenmesh-collector-setup.sh (hunk 3 above)
Comment (tartale) [reply scripts/aspenmesh-collector-setup.sh:22]: Suggestion to make this less obvious:

```suggestion
endpoint="${AM_METRICS_ENDPOINT:-https://metrics.cloud.aspenmesh.io/write}"
```

**Comment by tartale** [comment]
outdated; moved to a new repo

**Pull request closed without merging**